packages/bff/src/routers/catalogRouter.ts

```diff
@@ -7,21 +8,47 @@ import {
 } from "pagopa-interop-commons";
 import { api } from "../model/generated/api.js";
 import { PagoPAInteropBeClients } from "../providers/clientProvider.js";
+import { catalogServiceBuilder } from "../services/catalogService.js";
+import { toEserviceCatalogProcessQueryParams } from "../model/api/apiConverter.js";
+import { makeApiProblem } from "../model/domain/errors.js";
+import { bffGetCatalogErrorMapper } from "../utilities/errorMappers.js";
+import { fromBffAppContext } from "../utilities/context.js";
```

Comment:
this work was done by `contextMiddleware` [here](https://github.com/pagopa/interop-be-monorepo/blob/0ba01af7247eb48815ebb461a26215f5050614d1/packages/commons/src/context/context.ts#L34),  if it not found in headers it will be generated. 
@galales  It's necessary to avoid the reading of header `x-correlation-id` for all BFF routes?

Reply:
yes, BFF should ignore correlation id in the request.
The reason is it can be (maliciously or not) forced to a fixed or repeated value, which nullify the utility of the correlation id.
Right after the refactor we will remove the header from the openapi spec

Reply:
👍🏻 With this [commit](https://github.com/pagopa/interop-be-monorepo/pull/665/commits/29c354223b00f91f20ac0842078fc752715a9f2d) we generate a new correlationId when bff app context was built.

Reply:
@MalpenZibo some changes done in `fromBffAppContext` function for this purpose

Reply:
probably the best place to override this value is the `contextMiddleware` (potentially, calling multiple times the `fromBffAppContext` would generate different correlation ids)

Reply:
`fromBffAppContext` should be called every time a request is handled, it provide context for service's methods, so it works only once per request and shouldn't be called multiple time.

Reply:
I understand, however setting a value in the context with the middleware, and then replace it only when reading the existing context seems a code smell.
I prefer having the correct value directly in our source of truth

Reply:
As discussed offline, this improvement will be addressed in https://github.com/pagopa/interop-be-monorepo/pull/693.